viam-cartographer.go

```diff
@@ -747,9 +750,24 @@ func (cartoSvc *CartographerService) getNextDataPoint(ctx context.Context, lidar
 // DoCommand receives arbitrary commands.
 func (cartoSvc *CartographerService) DoCommand(ctx context.Context, req map[string]interface{}) (map[string]interface{}, error) {
 	if cartoSvc.closed {
-		cartoSvc.logger.Warn("DoCommand called after closed")
 		return nil, ErrClosed
 	}
+
+	if cartoSvc.modularizationV2Enabled {
+		return cartoSvc.doCommandModularizationV2(ctx, req)
+	}
+
+	return nil, viamgrpc.UnimplementedError
+}
+
+func (cartoSvc *CartographerService) doCommandModularizationV2(
+	ctx context.Context,
+	req map[string]interface{},
+) (map[string]interface{}, error) {
+	if _, ok := req["job_done"]; ok {
+		return map[string]interface{}{"job_done": cartoSvc.jobDone}, nil
+	}
+
```

Comment:
I thought it would be useful to keep full mod changes within the feature flag for organizational purposes but I can take it outside if that seems better